packages/blade/src/components/Input/_decisions/decisions.md

```diff
@@ -0,0 +1,212 @@
+| suffixContent                                                                                      | `string`                                                                  | No                                                                                  |                                                           | Used when we want to add suffix to the input field. Eg: `.00` or `@gmail.com`. If `showClearButton` is `true` then suffix will be placed before clear button. If trailing icon is provided then suffix will be placed before trailing icon(**Discuss if this is possible**)           |
+| isLoading                                                                                          | `boolean`                                                                 | No                                                                                  | `false`                                                   | Used to show a spinner when the the data is being loaded. **Discuss the position of the loader. Check options mentioned in the open questions below**                                                                                                                                 |
+| maxCharacters                                                                                      | `number`                                                                  | No                                                                                  |                                                           | Used to show a character counter under the input field                                                                                                                                                                                                                                |
+| validationState                                                                                    | `error` , `success`                                                       | No                                                                                  |                                                           | Applies the style to the input field based on validation status and shows either `errorText` or `successText` respectively                                                                                                                                                            |
```

Comment:
validationState will be error or success right? and If not present (undefined) it will be nothing. 

We are also doing the same on Checkbox. That if validationState is undefined we set nothing. 

Why not have a common value for that "undefined"? Maybe like `idle` or `default`

Reply:
`undefined` for a state is a good enough value, what's the benefit of adding an extra value?

Reply:
why do we need a value for that? if it doesn't exist it's not there. and for cases where you want to set it to none we can do this always right

```
validationState={text.lenght < 10: 'error' : undefined}
```

will adding another state will make sense?

Reply:
undefined is fine. Just that we are implicitly agreeing that undefined is a value, it's a bit harder to communicate the intent with undefined.